cupy/cuda/device.pyx

```diff
@@ -112,7 +116,10 @@ cdef class Device:
         if device is None:
             self.id = runtime.getDevice()
         else:
-            self.id = int(device)
+            try:
+                self.id = int(device)
+            except ValueError:
+                self.id = runtime.deviceGetByPCIBusId(str(device))
```

Comment:
How about implementing a `@staticmethod` `Device.from_pci_bus_id()`?
It's better to keep `__init__` simple.

Reply:
Yep, that's a better idea!

Reply:
I used a classmethod as this is inheritable